### DISCUSSION TITLE
Manually averaging metrics when logging

### DISCUSSION THREAD

**Discussion by digital-idiot** [Lightning-AI/pytorch-lightning#12636] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I have a metric from `torchmetric` as follows:

```python
Accuracy(
    num_classes=self.model.out_channels,
     average='none',
     ignore_index=self.ignore_index
)
```
Obviously I can not log this, however I don't want to set average to any aggregation. I want to log its mean in `training_step` but want to preserve the class wise metric to till end of epoch where I display it to terminal. I want the metric to reset at epoch end only, so can't call `compute()` in training step.

How to solve this?

**Comment by digital-idiot** [comment, marked as answer]
From [this comment](https://github.com/PyTorchLightning/pytorch-lightning/issues/4396#issuecomment-717571905) I had the idea that:
> if the .compute()method is called the internal state is reset.

However, this behavior has changed, now calling `compute()` does not reset the state of the metrics. See [PR #5409](https://github.com/PyTorchLightning/pytorch-lightning/pull/5409)
